### ISSUE TITLE
LoadError: UndefRefError: access to undefined reference

### ISSUE THREAD

**Issue by shkulkar20** [JuliaIO/JLD2.jl#252]
I get the following error when trying to load a file. I seem to have the required packages. Any insights on how to resolve this will be very helpful!

In[43]: using JLD2

In[44]: using FileIO

In[50]: @load "rock.jld2"
LoadError: UndefRefError: access to undefined reference
in expression starting at In[50]:1

Stacktrace:
 [1] getproperty at .\Base.jl:33 [inlined]
 [2] keys(::JLD2.JLDFile{JLD2.MmapIO}) at C:\Users\shkulkar\.julia\packages\JLD2\4AJn8\src\JLD2.jl:334
 [3] @load(::LineNumberNode, ::Module, ::Any, ::Vararg{Any,N} where N) at C:\Users\shkulkar\.julia\packages\JLD2\4AJn8\src\loadsave.jl:101
 [4] include_string(::Function, ::Module, ::String, ::String) at .\loading.jl:1091
 [5] execute_code(::String, ::String) at C:\Users\shkulkar\.julia\packages\IJulia\rWZ9e\src\execute_request.jl:27
 [6] execute_request(::ZMQ.Socket, ::IJulia.Msg) at C:\Users\shkulkar\.julia\packages\IJulia\rWZ9e\src\execute_request.jl:86
 [7] #invokelatest#1 at .\essentials.jl:710 [inlined]
 [8] invokelatest at .\essentials.jl:709 [inlined]
 [9] eventloop(::ZMQ.Socket) at C:\Users\shkulkar\.julia\packages\IJulia\rWZ9e\src\eventloop.jl:8
 [10] (::IJulia.var"#15#18")() at .\task.jl:356

**Comment by JonasIsensee** [comment]
Hi @shkulkar20,

`access to undefined reference` errors are usually not a problem of not having loaded all the packages.
Sadly this means that something in the file is broken.

You can try to recover your data piece by piece by explictly loading variables e.g. `@load "rock.jld2" myvariablename1`

Can you tell me what julia version and what JLD2 version you are using? Which versions where used to write the file?

**Comment by shkulkar20** [comment]
My Julia version is 1.5.2
The file was provided by the course I am enrolled in.
The following however seems to have worked (in case it provides a clue about what was wrong).

# This will overwrite your existing Project/Manifest files, so you might want to do it a new directory
download("https://storage.googleapis.com/mynerva-testing-static/rock_paper_scissors_classification-b07331ffb5e81744a7c1c7413524c5dc117d723c-Project.toml", "Project.toml")
download("https://storage.googleapis.com/mynerva-testing-static/rock_paper_scissors_classification-ddcbce466fc6ef64a8af59dfaa9c9196d2cd85a8-Manifest.toml", "Manifest.toml")
using Pkg
Pkg.instantiate()

# Restart the Julia session before running this
using Images, JLD2
load("rock.jld2")

**Comment by JonasIsensee** [comment]
Hm, I followed your instructions and additionally downloaded the `rock.jld` file (I think) from
https://storage.googleapis.com/mynerva-testing-static/rock_paper_scissors_classification-a63c641bba21050e27b94c58568347a566ef662b-rock.jld2

But for me the file loads just fine. I also tested JLD2 version `v0.2.0`, `v0.2.1`, `v0.2.2`, `v0.2.3`, and `v0.2.4`.
I also had a look at the file more closely. Since it only contains a single array and no other special structure I find it quite unlikely that you found an actual bug in JLD2. It may be that during downloading your file was corrupted e.g. only a partial download.

Do I understand correctly that now you cannot produce this error anymore yourself?
